<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Stepwise detection of recombination breakpoints using Phylpro — phylpro • stepwise</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Stepwise detection of recombination breakpoints using Phylpro — phylpro" />
<meta property="og:description" content="Stepwise detection of recombination breakpoints using phylogenetic profiling (Phylpro) at each step" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">stepwise</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Stepwise detection of recombination breakpoints using Phylpro</h1>
    
    <div class="hidden name"><code>phylpro.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Stepwise detection of recombination breakpoints using phylogenetic profiling (Phylpro) at each step</p>
    </div>

    <pre class="usage"><span class='fu'>phylpro</span>(<span class='no'>input_file</span>, <span class='no'>breaks</span>, <span class='no'>winHalfWidth</span>, <span class='no'>permReps</span>)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>input_file</th>
      <td><p>character string indicating the name of a Phylip format data input file</p></td>
    </tr>
    <tr>
      <th>breaks</th>
      <td><p>an integer vector of ordered site(s) just before the previously declared breakpoints</p></td>
    </tr>
    <tr>
      <th>winHalfWidth</th>
      <td><p>the window half width to use</p></td>
    </tr>
    <tr>
      <th>permReps</th>
      <td><p>the number of Monte Carlo replicates to use for the permutation distribution</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>The phylpro function implements phylogenetic profiling (Phylpro)
for detecting recombination breakpoints
(Weiller 1998) using a moving window of fixed width.
Breakpoints detected in previous steps of a
stepwise search may be conditioned upon.</p>
<p>For a given position of the moving window on the sequence alignment,
and for a given &#8220;target&#8221; sequence, a correlation is
computed to compare two distance vectors: the distance between the
target sequence and all other sequences
in the left half-window and the
distance between the target sequence and all others in the
right half-window. The pair-wise distance measure used is the proportion
of sites at which the sequences differ.
Discordance between the two distance vectors may
reflect a recombination event, located at the window centre,
in the history of the target sequence.
The minimum correlation over all target sequences
is regarded as a summary of the evidence for recombination at
the window centre. The individual correlations for the target
sequences may
also be of interest for suggesting sequence segments that descend from
historical recombination events.
Significance of observed correlation statistics
is assessed by a Monte Carlo permutation test.
When conditioning on breakpoints proposed at previous steps of
a stepwise search, permutation is restricted to sites within blocks
defined by the previously proposed breakpoints, as described by
Graham et al. (2004).</p>
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p></p>
<dt>polyposn</dt><dd><p>The site numbers of all ungapped polymorphic sites in the alignment</p></dd>
  <dt>corrs</dt><dd><p>Observed correlations that exceed the 90th percentile of the permutation null distribution</p></dd>
  <dt>winlocs</dt><dd><p>Window centres corresponding to the correlations in <code>corrs</code></p></dd>
  <dt>target.seqs</dt><dd><p>The target sequence that lead to a significant correlation in <code>corrs</code></p></dd>
  <dt>quants</dt><dd><p>90th and 95th percentiles of the permutation distribution</p></dd>

    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

     

<p>Graham J, McNeney B and Seillier-Moiseiwitsch F (2004). Stepwise detection of
recombination breakpoints in sequence alignments. <em>Bioinformatics</em> Sep 23;
[Epub ahead of print]</p>
<p>Weiller G (1998). Phylogenetic profiles: A graphical method for detecting
genetic recombination in homologous sequences.
<em>Mol Biol Evol</em>, <b>15</b>:326-335.</p>
<p><a href='http://stat-db.stat.sfu.ca/stepwise'>http://stat-db.stat.sfu.ca/stepwise</a></p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code>summary.phylpro</code>, <code>maxchi</code></p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'> <span class='fu'><a href='https://rdrr.io/r/utils/demo.html'>demo</a></span>(<span class='no'>phylpro</span>)</div><div class='output co'>#&gt; 
#&gt; 
#&gt; 	demo(phylpro)
#&gt; 	---- ~~~~~~~
#&gt; 
#&gt; &gt; inputFile&lt;-system.file("extdata", "simulinfile.txt", package = "stepwise")
#&gt; 
#&gt; &gt; breaks&lt;-c(548, 735, 832)
#&gt; 
#&gt; &gt; WinHalfWidth&lt;-30
#&gt; 
#&gt; &gt; permReps&lt;-10
#&gt; 
#&gt; &gt; b&lt;-phylpro(inputFile, breaks, WinHalfWidth, permReps)
#&gt; 
#&gt; &gt; summary.phylpro(b)
#&gt; --------------------------------------------------
#&gt; There were 20 site-specific Minimum Correlation statistics significant at the
#&gt; 10 percent level (10th percentile = -0.460, 5th percentile = -0.460):
#&gt; 
#&gt; Number Location  MinCor   targets
#&gt;      1      513  -0.549*   sample28__
#&gt;      2      526  -0.656*   sample28__
#&gt;      3      527  -0.627*   sample28__
#&gt;      4      528  -0.583*   sample28__
#&gt;      5      531  -0.648*   sample28__
#&gt;      6      548  -0.696*   sample28__
#&gt;      7      549  -0.691*   sample28__
#&gt;      8      556  -0.684*   sample28__
#&gt;      9      558  -0.635*   sample28__
#&gt;     10      564  -0.631*   sample28__
#&gt;     11      568  -0.625*   sample28__
#&gt;     12      573  -0.641*   sample28__
#&gt;     13      576  -0.528*   sample28__
#&gt;     14      579  -0.539*   sample28__
#&gt;     15      583  -0.536*   sample28__
#&gt;     16      588  -0.616*   sample28__
#&gt;     17      604  -0.509*   sample28__
#&gt;     18      616  -0.475*   sample28__
#&gt;     19      735  -0.492*   sample29__
#&gt;     20      744  -0.460*   sample29__
#&gt; --------------------------------------------------
#&gt; Notes - "Location" is the polymorphic site just before the proposed breakpoint.
#&gt;       - MinCor statistics significant at the 5 percent level indicated by a * 
#&gt; </div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Brad McNeney.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


